src/LanguageServerManager.ts

```diff
@@ -168,75 +223,34 @@ export class LanguageServerManager {
             //disable the simple providers (the language server will handle all of these)
             this.disableSimpleProviders();
 
-            // The server is implemented in node
-            let serverModule = this.context.asAbsolutePath(
-                path.join('dist', 'LanguageServerRunner.js')
-            );
-
-            //give the runner the specific version of bsc to run
-            const args = [
-                this.selectedBscInfo.path,
-                (this.context.extensionMode === vscode.ExtensionMode.Development).toString()
-            ];
-            // If the extension is launched in debug mode then the debug server options are used
-            // Otherwise the run options are used
-            let serverOptions: ServerOptions = {
-                run: {
-                    module: serverModule,
-                    transport: TransportKind.ipc,
-                    args: args
-                },
-                debug: {
-                    module: serverModule,
-                    transport: TransportKind.ipc,
-                    args: args,
-                    // --inspect=6009: runs the server in Node's Inspector mode so VS Code can attach to the server for debugging
-                    options: { execArgv: ['--nolazy', '--inspect=6009'] }
-                }
-            };
-
-            // Options to control the language client
-            let clientOptions: LanguageClientOptions = {
-                // Register the server for various types of documents
-                documentSelector: [
-                    { scheme: 'file', language: 'brightscript' },
-                    { scheme: 'file', language: 'brighterscript' },
-                    { scheme: 'file', language: 'xml' }
-                ],
-                synchronize: {
-                    // Notify the server about file changes to every filetype it cares about
-                    fileEvents: workspace.createFileSystemWatcher('**/*')
-                }
-            };
+            this.client = this.constructLanguageClient();
 
-            // Create the language client and start the client.
-            this.client = new LanguageClient(
-                'brighterScriptLanguageServer',
-                LANGUAGE_SERVER_NAME,
-                serverOptions,
-                clientOptions
-            );
-            // Start the client. This will also launch the server
-            this.clientDispose = this.client.start();
-            await this.client.onReady();
             this.client.onDidChangeState((event: StateChangeEvent) => {
```

Comment:
This was the primary fix. We need to subscribe to these events _before_ starting the client so we pick up the `starting` and `running` states.